v3/src/components/graph/utilities/graph-utils.ts

```diff
@@ -494,6 +495,36 @@ export const leastSquaresLinearRegression = (iValues: Point[], iInterceptLocked:
   return tRegression
 }
 
+interface ISumOfSquares {
+  cellKey: Record<string, string>
+  dataConfig: IGraphDataConfigurationModel
+  intercept: number
+  slope: number
+  defaultVal?: number
+}
+
+export const calculateSumOfSquares = ({ cellKey, dataConfig, defaultVal=0, intercept, slope }: ISumOfSquares) => {
+  const dataset = dataConfig?.dataset
+  const caseData = dataset?.cases
+  const xAttrID = dataConfig?.attributeID("x") ?? ""
+  const yAttrID = dataConfig?.attributeID("y") ?? ""
+  let sumOfSquares = 0
+  caseData?.forEach((datum: any) => {
+    const fullCaseData = dataConfig?.dataset?.getCase(datum.__id__)
+    if (fullCaseData && dataConfig?.isCaseInSubPlot(cellKey, fullCaseData)) {
+      const x = dataset?.getNumeric(datum.__id__, xAttrID) ?? defaultVal
+      const y = dataset?.getNumeric(datum.__id__, yAttrID) ?? defaultVal
```

Comment:
I would think that `NaN` is always the preferable default so you wouldn't need the argument.